src/tileset/wangset.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+
+use xml::attribute::OwnedAttribute;
+
+use crate::{
+    error::Error,
+    properties::{parse_properties, Properties},
+    util::{get_attrs, parse_tag, XmlEventResult},
+    Result, TileId,
+};
+
+mod wang_color;
+pub use wang_color::WangColor;
+mod wang_tile;
+pub use wang_tile::{WangId, WangTile};
+
+/// Undocummented WangSet types
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum WangSetType {
+    Corner,
+    Edge,
+    Mixed,
+}
+
+impl Default for WangSetType {
+    fn default() -> Self {
+        WangSetType::Mixed
+    }
+}
+
+/// Raw data belonging to a WangSet.
+#[derive(Debug, PartialEq, Clone)]
+pub struct WangSet {
+    /// The name of the Wang set
+    pub name: String,
+    /// Type of wangset
+    pub wang_type: WangSetType,
+    /// The tile ID of the tile representing this Wang set.
+    pub tile: Option<TileId>,
+    /// A color that can be used to define the corner and/or edge of a Wang tile.
+    pub wang_colors: Vec<WangColor>,
+    /// A color that can be used to define the corner and/or edge of a Wang tile.
+    pub wang_tiles: HashMap<TileId, WangTile>,
```

Comment:
I wonder if this couldn't be just `HashMap<TileId, WangId>`, since the `WangTile` is just going to contain the same `TileId` again, right?

We might not need a `WangTile` struct at all. In Tiled it only exists temporarily as a way to prepare the data for saving in a consistent order.

The documentation string still needs to be adjusted.

Reply:
Well, WangTile describes the tile connections that a single Tile has on that Wang Set. If you delete that 8 size array, not sure you can get that data again. The only simplification I see we could do is to delete the TileId value that is hold inside WangTile struct, as it can be recovered from the wang_tiles HashMap. I just added there as it's very small in size and can be useful to have.

Reply:
Yes, the `WangTile` is just tile ID + `WangId` currently. If you remove tile ID, all that left is the `WangId`, in which case why would you have a wrapping `WangTile` struct at all? That's why I was suggesting to remove it.

Reply:
Well, I would prefer to keep the wrapping struct just to implement FromStr and be able to return an encoding error. Seems more clear to me.

Reply:
Alright, we can keep the struct if that is more convenient, but do we need to actually use it in the `wang_tiles` hash map, or could we keep the `WangTile` for parsing and error reporting, but store just the `WangId` there? In general in other hash maps, we have avoided storing the key as part of the value as well.

Reply:
Leaving like this?:

```rust
pub struct WangTile {
    /// The Wang ID,
    pub wang_id: WangId,
}
`

Reply:
> Well, I would prefer to keep the wrapping struct just to implement FromStr and be able to return an encoding error. Seems more clear to me.
> 
> Also, if there is more data inserted on this field in the future, we already have the struct prepared.

Seems wiser to make the hashmap private, change the value type and expose an access method that returns the wrapping struct, as done everywhere else in the crate (i.e. Data vs wrapper structures)

Reply:
> Alright, we can keep the struct if that is more convenient, but do we need to actually use it in the `wang_tiles` hash map, or could we keep the `WangTile` for parsing and error reporting, but store just the `WangId` there? In general in other hash maps, we have avoided storing the key as part of the value as well.

Exactly this, as done with tilesets & their stored tiles, for instance. Store raw/minimal data in hashmap, return more convenient wrapper to user.

Reply:
> 

About this, I don't see any advantage on returning, for example, an iterator in the same way that we return tiles in Tileset, as no new info will be added to WangTile struct.